go/pkg/fakes/server.go

```diff
@@ -279,6 +279,62 @@ func (f *OutputFile) Apply(ac *repb.ActionResult, s *Server, execRoot string) er
 	return nil
 }
 
+// OutputDir is to be added as an output of the fake action.
+type OutputDir struct {
+	Path string
+}
+
+// Apply puts the file in the fake CAS and the given ActionResult.
+func (d *OutputDir) Apply(ac *repb.ActionResult, s *Server, execRoot string) error {
+	root, ch, err := BuildDir(d.Path, s, execRoot)
+	if err != nil {
+		return fmt.Errorf("failed to build directory tree: %v", err)
+	}
+	tr := &repb.Tree{
+		Root:     root,
+		Children: ch,
+	}
+	treeBlob, err := proto.Marshal(tr)
+	if err != nil {
+		return fmt.Errorf("failed to marshal tree: %v", err)
+	}
+	treeDigest := s.CAS.Put(treeBlob)
+	ac.OutputDirectories = append(ac.OutputDirectories, &repb.OutputDirectory{Path: d.Path, TreeDigest: treeDigest.ToProto()})
+	return nil
+}
+
+// Builds the directory tree by recursively iterating through the directory.
+func BuildDir(path string, s *Server, execRoot string) (root *repb.Directory, childDir []*repb.Directory, err error) {
```

Comment:
Maybe add comments that it looks like `tree.go` `ComputeMerkleTree`, but adapted?